gpm_api/tests/test_utils/test_slices.py

```diff
@@ -0,0 +1,481 @@
+    slices = [slice(0, 3), slice(1, 6), slice(8, 9)]
+    expected_indices = [0, 1, 2, 3, 4, 5, 8]
+    returned_indices = gpm_slices.get_indices_from_list_slices(slices, check_non_intersecting=False)
+    assert np.array_equal(returned_indices, expected_indices)
+
+    # Test with intersection check
+    with pytest.raises(ValueError):
+        gpm_slices.get_indices_from_list_slices(slices, check_non_intersecting=True)
+
+    # Test with empty list
+    assert gpm_slices.get_indices_from_list_slices([]).size == 0
+
+
+def test_list_slices_intersection() -> None:
+    """Test list_slices_intersection"""
+
+    slices = [
+        [slice(0, 3), slice(4, 7)],  # 0 1 2|  4 5 6|
+        [slice(2, 5), slice(5, 8)],  #     2 3 4|5 6 7
+    ]  #                 intersection:     2|3 4|5 6|
+    # Common indices: 2, 4, 5, 6
+    expected_list = [slice(2, 3), slice(4, 5), slice(5, 7)]
+    returned_list = gpm_slices.list_slices_intersection(*slices)
+    assert returned_list == expected_list
+
+    # Hole in one list: should not be patched
+    slices = [
+        [slice(0, 10)],
+        [slice(0, 5), slice(5, 10)],
+    ]
+    expected_list = [slice(0, 5), slice(5, 10)]
+    returned_list = gpm_slices.list_slices_intersection(*slices)
+    assert returned_list == expected_list
+
+    # Repeated slices
+    slices = [
+        [slice(0, 3), slice(3, 6)],
+        [slice(0, 3), slice(3, 6)],
+    ]
+    expected_list = [slice(0, 3), slice(3, 6)]
+    returned_list = gpm_slices.list_slices_intersection(*slices)
+    assert returned_list == expected_list
+
+    # Non-intersecting lists
+    slices = [
+        [slice(0, 3)],
+        [slice(4, 7)],
+    ]
+    expected_list = []
+    returned_list = gpm_slices.list_slices_intersection(*slices)
+    assert returned_list == expected_list
+
+    # Test with empty list
+    assert gpm_slices.list_slices_intersection() == []
+
+
+def test_list_slices_union() -> None:
+    """Test list_slices_union"""
+
+    slices = [
+        [slice(0, 3), slice(4, 7)],
+        [slice(4, 7), slice(7, 10)],
+    ]
+    expected_list = [slice(0, 3), slice(4, 10)]
+    returned_list = gpm_slices.list_slices_union(*slices)
+    assert returned_list == expected_list
+
+    # Test with one empty list
+    slices[1] = []
+    expected_list = [slice(0, 3), slice(4, 7)]
+    returned_list = gpm_slices.list_slices_union(*slices)
+    assert returned_list == expected_list
+
+
+def test_list_slices_difference() -> None:
+    """Test list_slices_difference"""
+
+    # Base cases
+    assert gpm_slices.list_slices_difference([slice(3, 6)], [slice(1, 2)]) == [slice(3, 6)]
+    assert gpm_slices.list_slices_difference([slice(3, 6)], [slice(1, 4)]) == [slice(4, 6)]
+    assert gpm_slices.list_slices_difference([slice(3, 6)], [slice(1, 8)]) == []
+    assert gpm_slices.list_slices_difference([slice(3, 6)], [slice(5, 8)]) == [slice(3, 5)]
+    assert gpm_slices.list_slices_difference([slice(3, 6)], [slice(7, 8)]) == [slice(3, 6)]
```

Comment:
Add here also test case `[slice(3, 6)], [slice(6, 8)]) == [slice(3, 6)]` to test when stop and start concides ;)